plugins/hls-hlint-plugin/hls-hlint-plugin.cabal

```diff
@@ -62,7 +62,7 @@ library
     , ghc-lib-parser
     , ghc-lib-parser-ex
   if impl(ghc >= 9.2)
-    build-depends: apply-refact ^>= 0.11.0.0
```

Comment:
Why isn't the version selection handled by Cabal? Just set proper bounds for the `ghc` dependency  in the `apply-refact` Cabal descriptor and remove all these conditionals from here

Reply:
Like this? Now we just explicitly forbid 0.11.0

Reply:
No, simply: `build-depends: apply-refact`. Then cabal will select the version that works with the `ghc` package that is built-in

Reply:
That would in theory permit version 0.11.0, I want to explicitly forbid this. Or do you think that's pointless?

Reply:
I think it's pointless, as it won't be used in any scenarios. If you still have concerns, mark it as deprecated in Hackage